addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/ahamodel/SwitchModel.java

```diff
@@ -40,7 +43,11 @@ public void setState(BigDecimal state) {
     }
 
     public String getMode() {
-        return mode;
+        if (mode.equals("auto")) {
```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)